### PR TITLE
enhance: Query slot for compaction task

### PR DESCRIPTION
**Pull request by bigsheeper**
Query slot of compaction in datanode, and transfer the control logic for limiting compaction tasks from datacoord to the datanode.

issue: https://github.com/milvus-io/milvus/issues/32809

### CODE REVIEW

**Review by XuanYang-cn — CHANGES_REQUESTED** [review]

File: internal/datacoord/compaction_scheduler.go [hunk 1]
```diff
@@ -211,3 +217,50 @@ func (s *CompactionScheduler) LogStatus() {
...
+			}
+			mu.Lock()
+			defer mu.Unlock()
+			nodeSlots[nodeID] = resp.GetNumSlots()
+		}(nodeID)
+	}
+	wg.Wait()
+	log.Debug("query slot done", zap.Any("nodeSlots", nodeSlots))
+	return nodeSlots
+}
```
[the remark this answers is not in the document]
Comment (bigsheeper) [reply internal/datacoord/compaction_scheduler.go]: done

**Comment by bigsheeper** [comment]
rerun ut

File: internal/datacoord/compaction_scheduler.go [hunk 2]
```diff
@@ -62,22 +64,27 @@ func (s *CompactionScheduler) Submit(tasks ...*compactionTask) {
 
 // Schedule pick 1 or 0 tasks for 1 node
 func (s *CompactionScheduler) Schedule() []*compactionTask {
-	nodeTasks := make(map[int64][]*compactionTask) // nodeID
-
 	s.taskGuard.Lock()
```
Comment (czs007) [inline internal/datacoord/compaction_scheduler.go:67]: Rlock

File: internal/proto/data_coord.proto [hunk 3]
```diff
@@ -828,3 +830,10 @@ message GcControlRequest {
   GcCommand command = 2;
   repeated common.KeyValuePair params = 3;
 }
+
+message QuerySlotRequest {}
+
+message QuerySlotResponse {
+  common.Status status = 1;
+  int64 num_slots = 2;
```
Comment (czs007) [inline internal/proto/data_coord.proto:838]: add nodeID

**Pull request merged**